### PR TITLE
Fix host_height, host_timestamp return value 

### LINKED ISSUE #242: Change host_height, host_timestamp return value to a Result in ClientReader, ConnectionReader, ChannelReader and ValidationContext
## Summary

For downstream implementation of `ibc-rs` interface, `host_height`, `host_timestamp` may be wrong, here we suggest to return a Result type.

## Acceptance Criteria

change before is 
```rust
fn host_height(&self) -> Height;
fn host_timestamp(&self) -> Timestamp;
```
change after is 
```rust
fn host_height(&self) -> Result<Height, Error>;
fn host_timestamp(&self) -> Result<Timestamp, Error>;
```
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by DaviRain-Su**
Closes: #242 

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: .changelog/unreleased/features/242-change-return-value.md
Comment (plafer) [inline .changelog/unreleased/features/242-change-return-value.md]:
```suggestion
- Change `host_height`, `host_timestamp` return value to a `Result` in `ClientReader`, `ConnectionReader`, and `ChannelReader`
```

Comment (plafer) [inline .changelog/unreleased/features/242-change-return-value.md]: Additionally, can we move this to the `.changelog/unreleased/breaking-changes` directory?

**Review by plafer — CHANGES_REQUESTED** [review]
Looks great, thanks! Just 2 minor changes to the changelog entry, and we'll get that merged in!

**Comment by plafer** [comment]
Also, now that #241 is merged, could you apply the same change to `ValidationContext`?

**Comment by DaviRain-Su** [comment]
> Also, now that #241 is merged, could you apply the same change to `ValidationContext`?

Done

File: crates/ibc/src/core/context.rs
Comment (plafer) [inline crates/ibc/src/core/context.rs:96]: `ClientError` is not appropriate here, but we don't have a good solution for that at the moment. We would really need some `HostError`. We can fix this with #164.

**Review by plafer — APPROVED** [review]
🙏

**Review by plafer — CHANGES_REQUESTED** [review]
Will re-approve after merge conflicts are resolved

File: crates/ibc/src/core/context.rs
Comment (DaviRain-Su) [reply crates/ibc/src/core/context.rs:96]: I would also like to use this way to solve, it is not very clear that the current no_std support is not very perfect

**Review by plafer — APPROVED** [review]

**Pull request merged**
